cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```diff
@@ -86,32 +168,81 @@ public void schedule(Id.Program program, SchedulableProgramType programType, Ite
 
   @Override
   public List<String> getScheduleIds(Id.Program program, SchedulableProgramType programType) {
-    return ImmutableList.of();
+    Set<String> scheduleIds;
+    synchronized (this) {
+      scheduleIds = ImmutableSet.copyOf(scheduleSubscribers.keySet());
+    }
+    ImmutableList.Builder<String> builder = ImmutableList.builder();
+    String programScheduleId = getProgramScheduleId(program, programType);
+    for (String scheduleId : scheduleIds) {
+      if (scheduleId.startsWith(programScheduleId)) {
+        builder.add(scheduleId);
+      }
+    }
+    return builder.build();
   }
 
   @Override
   public void suspendSchedule(Id.Program program, SchedulableProgramType programType, String scheduleName) {
-
+    StreamSubscriber subscriber = scheduleSubscribers.get(getScheduleId(program, programType, scheduleName));
+    if (subscriber != null) {
+      subscriber.suspendScheduleTask(program, programType, scheduleName);
+    }
   }
 
   @Override
   public void resumeSchedule(Id.Program program, SchedulableProgramType programType, String scheduleName) {
-
+    StreamSubscriber subscriber = scheduleSubscribers.get(getScheduleId(program, programType, scheduleName));
+    if (subscriber != null) {
+      subscriber.resumeScheduleTask(program, programType, scheduleName);
+    }
   }
 
   @Override
   public void deleteSchedule(Id.Program programId, SchedulableProgramType programType, String scheduleName) {
-
+    StreamSubscriber subscriber = scheduleSubscribers.remove(getScheduleId(programId, programType, scheduleName));
+    if (subscriber != null) {
+      subscriber.deleteSchedule(programId, programType, scheduleName);
+      synchronized (this) {
+        if (subscriber.isEmpty()) {
+          subscriber.cancel();
+          Id.Stream streamId = subscriber.getStreamId();
+          streamSubscribers.remove(streamId);
+        }
+      }
+    }
   }
 
   @Override
   public void deleteSchedules(Id.Program programId, SchedulableProgramType programType) {
-
+    Set<String> scheduleIds;
+    synchronized (this) {
```

Comment:
again it is not clear how synchronizing helps here. If you don't synchronize the following code, then the subscribers can still get updated concurrently.